utf8proc.c

```diff
@@ -165,12 +168,9 @@ UTF8PROC_DLLEXPORT utf8proc_ssize_t utf8proc_encode_char(utf8proc_int32_t uc, ut
     dst[0] = 0xC0 + (uc >> 6);
     dst[1] = 0x80 + (uc & 0x3F);
     return 2;
-  } else if (uc == 0xFFFF) {
-    dst[0] = 0xFF;
-    return 1;
-  } else if (uc == 0xFFFE) {
-    dst[0] = 0xFE;
-    return 1;
+  } else if (((utf8proc_uint32_t)uc)-0xd800 < 0x800) {
+    // Surrogates not valid code point, can't be encoded!
+    return 0;
```

Comment:
Except this isn't a validity checking step per se (I agree that the utf8proc_map function should check for surrogates), this is an encoding function.  The question is whether to barf if the user specifically requests to encode a surrogate char.

I just want to be careful about changing the behavior here.  This is a behavior change in an existing API, and whether it is a bug fix is a matter of perspective.